src/api/middleware/validators/validatorUtils.js

```diff
@@ -15,6 +15,29 @@ const valuesInSet = (set) => (arr) => {
     return true;
 };
 
+/**
+ * Returns a validator that checks whether all of the elements of an array belong to the provided set of values
+ * @param {Array} set
+ */
+const stringOrValuesInSet = (set) => (val) => {
+
+    const checkValid = (value) => (arr) => {
+        if (!arr.includes(value)) {
+            throw new Error(ValidationReasons.IN_ARRAY(set));
+        }
+    };
```

Comment:
This does not need to be a nested function, the return value of the outer function is never used more than once.

Reply:
I don't understand. the `checkValid` function is only used in that function, I abstracted it in order to avoid replication in the if block below

Reply:
I meant changing the function `checkValid` from being called like `checkValid(value)(arr)` to being invoked like `checkValid(value, arr)`

Reply:
Got it, done.